src/containers/AppBody.jsx

```diff
@@ -59,6 +54,29 @@ export class AppBody extends Component {
       cookie.load('orgs').split(' ').forEach(org => dispatch(addOrg(org)));
     }
 
+    this.loadSnippet();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Reload snippet if URL changes. Assume it is valid, and
+    // if it is not, the call to loadSnippet will handle it.
+    if (this.state.pathname !== nextProps.router.location.pathname) {
```

Comment:
Since you've wrapped the component using `withRouter`, you have access to the router from current props, so you can compare it with `nextProps` instead of having to save the pathname to state. So
```js
if (this.props.router.location.pathname !== nextProps.router.location.pathname) {
  // ...
}
```